cwltool/job.py

```diff
@@ -268,8 +308,12 @@ def _execute(self, runtime, env, rm_tmpdir=True, move_outputs="move", secret_sto
 
         if processStatus != "success":
             _logger.warning(u"[job %s] completed %s", self.name, processStatus)
+            if research_obj:
```

Comment:
Consider taking the inner blocks `if research_obj:` out of the `if status != success:` block to avoid code duplication.

Reply:
Thanks @psafont, good point. the new commit has resolved this.